manifests/ubuntu/agent6.pp

```diff
@@ -16,11 +16,14 @@
   Optional[String] $apt_keyserver = undef,
 ) inherits datadog_agent::params {
 
-  ensure_packages(['apt-transport-https'])
+  exec { 'apt-transport-https':
```

Comment:
I would like to continue using `ensure_packages` to be honest. I'm not sure how it would cause a cycle. If you're positive it's partly to blame we can go ahead and do this, but I'd rather not. Also, the line replaced should probably be:
```
ensure_packages(['apt-transport-https'], {'ensure' => 'present'})
```